.github/workflows/build_and_deploy.yml

```
@@ -361,44 +361,32 @@ jobs:
     needs: [config, build_and_deploy]
     runs-on: macos-12
     env:
+      IOS_X86_64_PATH: artifact/voicevox_core-x86_64-apple-ios
+      IOS_AARCH64_SIM_PATH: artifact/voicevox_core-aarch64-apple-ios-sim
+      IOS_AARCH64_PATH: artifact/voicevox_core-aarch64-apple-ios
       ASSET_NAME: voicevox_core-ios-xcframework-cpu-${{ needs.config.outputs.version }}
     steps:
       - uses: actions/checkout@v3
       - uses: actions/download-artifact@v2
         with:
           name: voicevox_core-x86_64-apple-ios
-          path: artifact/voicevox_core-x86_64-apple-ios
+          path: ${{ env.IOS_X86_64_PATH }}
       - uses: actions/download-artifact@v2
         with:
           name: voicevox_core-aarch64-apple-ios-sim
-          path: artifact/voicevox_core-aarch64-apple-ios-sim
+          path: ${{ env.IOS_AARCH64_SIM_PATH }}
       - uses: actions/download-artifact@v2
         with:
           name: voicevox_core-aarch64-apple-ios
-          path: artifact/voicevox_core-aarch64-apple-ios
-      - name: Create fat binary
-        run: |
-          mkdir -p "artifact/voicevox_core-sim"
-          lipo -create "artifact/voicevox_core-x86_64-apple-ios/libvoicevox_core.dylib" "artifact/voicevox_core-aarch64-apple-ios-sim/libvoicevox_core.dylib" -output "artifact/voicevox_core-sim/libvoicevox_core.dylib"
-      - name: Create XCFramework
-        run: |
-          mkdir -p "artifact/${{ env.ASSET_NAME }}"
-          # 必要なファイルだけコピー
-          mkdir -p "Headers-sim"
-          cp -v artifact/voicevox_core-x86_64-apple-ios/voicevox_core.h "Headers-sim"
-          cp -v crates/voicevox_core_c_api/xcframework/Headers/module.modulemap "Headers-sim"
-          mkdir -p "Headers-aarch64"
-          cp -v artifact/voicevox_core-aarch64-apple-ios/voicevox_core.h "Headers-aarch64"
-          cp -v crates/voicevox_core_c_api/xcframework/Headers/module.modulemap "Headers-aarch64"
-          xcodebuild -create-xcframework \
-            -library "artifact/voicevox_core-sim/libvoicevox_core.dylib" \
-            -headers "Headers-sim" \
-            -library "artifact/voicevox_core-aarch64-apple-ios/libvoicevox_core.dylib" \
-            -headers "Headers-aarch64" \
-            -output "artifact/${{ env.ASSET_NAME }}/voicevox_core.xcframework"
+          path: ${{ env.IOS_AARCH64_PATH }}
+      - name: Create xcframework
+        run:
+          build_util/make_ios_xcframework.bash
+        env:
+          OUTPUT_ASSET_PATH: artifact/voicevox_core-ios-xcframework-cpu
```

`OUTPUT_ASSET_PATH` の値 `artifact/voicevox_core-ios-xcframework-cpu` と全く同じ文字列が直後のステップでも使われているため、後ろのステップでも変数で参照できるようにするのが良いのかなと思いました。

`$GITHUB_OUTPUT` を使う方法：

```suggestion
      - name: Create xcframework
        id: create-xcframework
        run: |
          build_util/make_ios_xcframework.bash
          echo "output_asset_path=${OUTPUT_ASSET_PATH}" >> "$GITHUB_OUTPUT"
        env:
          OUTPUT_ASSET_PATH: artifact/voicevox_core-ios-xcframework-cpu
```

これで、後続のステップでは `${{ steps.create-xcframework.outputs.output_asset_path }}` でアーティファクトの位置を取得できると思います。記述が若干長いですが、どのステップで定義された変数なのかはわかりやすいです。

他に、`$GITHUB_ENV` に追加する方法もあります。 https://github.com/VOICEVOX/voicevox_core/pull/723#discussion_r1484280074 の議論でわかりづらいという意見もありましたが、個人的には `$GITHUB_ENV` を使っても良いと思います（むしろこの方が、単に `$OUTPUT_ASSET_PATH` と書けば文字列を取得できるのですっきり書けそうな気がします）。しかしどちらでも大丈夫だと思います。

```suggestion
      - name: Create xcframework
        run: |
          build_util/make_ios_xcframework.bash
          echo "OUTPUT_ASSET_PATH=${OUTPUT_ASSET_PATH}" >> "$GITHUB_ENV"
        env:
          OUTPUT_ASSET_PATH: artifact/voicevox_core-ios-xcframework-cpu
```

---

なるほど、こういう書き方があるのですね。
上の書き方を使って記述の重複を削除しました！